packages/expo-firebase-analytics/src/ExpoFirebaseAnalytics.ts

```diff
@@ -111,4 +118,8 @@ export default {
   async setDebugModeEnabled(isEnabled: boolean): Promise<void> {
     return callAnalyticsModule('setDebugModeEnabled', isEnabled);
   },
+  setClientId(clientId: string): void {
+    clientIdForJS = clientId;
+    maybeCreatePureJSAnalyticsTracker();
```

Comment:
i think this might cause problems if you call other methods like `setDebugModeEnabled` before `setClientId` - the debug mode state would be lost when we re-create the module

Reply:
good point. I added a method to mutate the value on the JS module instead, similar to how `setDebugModeEnabled` works but without going through `callAnalyticsModule` (since the method doesn't exist on the native modules)